inc/saibuffer.h

```diff
@@ -328,14 +328,23 @@ typedef enum _sai_buffer_profile_attr_t
      * default to 0. */
     SAI_BUFFER_PROFILE_ATTR_XOFF_TH,
 
-    /** set the buffer profile XON threshold in byte [sai_uint32_t]
+    /** set the buffer profile XON non-hysteresis threshold in byte [sai_uint32_t]
      * Valid only for ingress PG (CREATE_AND_SET)
-     * Generate XON when the total buffer usage of this PG
-     * is less this threshold and available buffer in the PG buffer
-     * is larger than the XOFF threahold.
+     * Ensures to generate XON, and ignore hysteresis, when the total 
+     * buffer usage of this PG is less than this threshold and available 
+     * buffer in the PG buffer is larger than the XOFF threshold.
      * default to 0. */
     SAI_BUFFER_PROFILE_ATTR_XON_TH,
 
+    /** set the buffer profile XON hysteresis threshold in byte [sai_uint32_t]
```

Comment:
agree with you. This is another mechanism to generate XON. The existing one compares the available PG buffer usage with a threshold to generate xon. The new one in below is an OFFSET, when the available PG buffer is less than XOFF threshold minus the OFFSET, it will generate the xon.

When a user sets both of them, the XON trigger threshold is max(XON_TH, XOFF_TH-XON_OFFSET)

It is probably more clear to rename the new attribute as  XON_OFFSET_TH.

@gmirek , can you add the equation into the comments so that user understand the relationship between the two XON threshold.